Formula/glib-utils.rb

```diff
@@ -0,0 +1,107 @@
+class GlibUtils < Formula
+  include Language::Python::Shebang
+
+  desc "Python utilities for GLib"
+  homepage "https://developer.gnome.org/glib/"
+  url "https://download.gnome.org/sources/glib/2.72/glib-2.72.2.tar.xz"
+  sha256 "78d599a133dba7fe2036dfa8db8fb6131ab9642783fc9578b07a20995252d2de"
+  license "LGPL-2.1-or-later"
+
+  depends_on "meson" => :build
+  depends_on "ninja" => :build
+  depends_on "pkg-config" => :build
+  depends_on "glib"
+  depends_on "python@3.10"
+
+  def install
+    # TODO: This is a workaround for `brew audit --new-formula`.
+    #       Use `patch` rather than `inreplace` (see also `glib`).
+    # replace several hardcoded paths with homebrew counterparts
+    inreplace "gio/xdgmime/xdgmime.c",
+              'xdg_data_dirs = "/usr/local/share/:/usr/share/";',
+              "xdg_data_dirs = \"#{HOMEBREW_PREFIX}/share/:/usr/local/share/:/usr/share/\";"
+    inreplace "glib/gutils.c",
+              'data_dirs = "/usr/local/share/:/usr/share/";',
+              "data_dirs = \"#{HOMEBREW_PREFIX}/share/:/usr/local/share/:/usr/share/\";"
+
+    # Point the headers and libraries to `glib`.
+    # The headers and libraries will be removed later because they are provided by `glib`.
+    glib = Formula["glib"]
+    args = std_meson_args.delete_if do |arg|
+      arg.start_with?("--includedir=", "--libdir=")
+    end
+    args += %W[
+      --includedir=#{glib.opt_include}
+      --libdir=#{glib.opt_lib}
+    ]
+
+    # Disable dtrace; see https://trac.macports.org/ticket/30413
+    # and https://gitlab.gnome.org/GNOME/glib/-/issues/653
+    args += %W[
+      --default-library=both
+      --localstatedir=#{var}
+      -Diconv=auto
+      -Dgio_module_dir=#{HOMEBREW_PREFIX}/lib/gio/modules
+      -Dbsymbolic_functions=false
+      -Ddtrace=false
+    ]
+
+    mkdir "build" do
+      system "meson", *args, ".."
+      system "ninja", "-v"
+
+      # Skip files already in glib
+      Formula["meson"].opt_libexec.cd do
+        system "bin/python3", "-c", pyscript
+      end
+      system "ninja", "install", "-v"
+    end
+
+    # Delete non python files because they are provided by `glib`
+    python_shebang_regex = %r{^#! ?/usr/bin/(?:env )?python(?:[23](?:\.\d{1,2})?)?( |$)}
+    shebang_max_length = 28 # the length of "#! /usr/bin/env pythonx.yyy "
+    prefix.find do |f|
+      next unless f.file?
+      next if f.extname == ".py" || python_shebang_regex.match?(f.read(shebang_max_length))
+
+      f.unlink
+    end
+
+    # Delete empty directories
+    # Note: We need to traversal the directories in reverse order (i.e. deepest first).
+    #       Also, we should put checking emptiness and deletion in a single loop.
+    #       `dirs.select(&:empty?).map(&:rmdir)` will not work because it will not delete
+    #       directories that only contain empty directories.
+    prefix.glob("**/").reverse_each { |d| d.rmdir if d.empty? }
```

Comment:
```suggestion
    prefix.find.reverse_each { |d| d.rmdir if d.empty? }
```
You can still use `Pathname#find` here.

Reply:
`prefix.find` also returns files and `f.empty?` can be true for an empty file. `prefix.glob("**/")` only returns directories because of the tailing slash in the glob pattern.

Reply:
@carlocab is @XuehaiPan's answer OK for you? If yes, please mark this part of the conversation as resolved so that we can proceed with a rebase + rebuild.